controller/lib/core/pinch_valve.cpp

```diff
@@ -212,7 +214,15 @@ void PinchValve::SetOutput(float value) {
   // The motor's zero position is at the home offset
   // which corresponds to fully open (i.e. 100% flow)
   // The valve is closed at a position of -max_move;
-
   float pos = (value - 1.0f) * max_move;
+
+  // If the commanded position changed since last cycle
+  // halt the previous move if it's still in progress.
+  // This makes the valve control much smoother when used
+  // within a servo loop
```

Comment:
I found the explanation you posted in slack to be really helpful -- based on just this I don't think I would understand why this is a good change.

To quote you with some minor edits:

> Once you put a move in motion you can't change the destination position until the move ends.  That means that if the servo loop commands a relatively large move of the valve in one cycle, then any new commands will be ignored until that move finishes.  This can cause the pinch valve  to oscillate if you turn the gains up too high.
>
> To fix this, we hard-stop the valve before sending a new position.  This doesn't do anything if the valve was already stopped, but if it was in the middle of a big move it tells it to abandon that old move and start a new move to the new commanded position.
>
> This seems to really smooth out the pinch valve motion and allow for higher gains.

Dunno if you'd be willing to put something more explicit in the code, up to just what I quoted above.